rust/operator/src/hdfs_controller.rs

```diff
@@ -364,58 +366,77 @@ fn rolegroup_statefulset(
     tracing::info!("Setting up StatefulSet for {:?}", rolegroup_ref);
     let service_name = rolegroup_ref.object_name();
 
+    // PodBuilder for StatefulSet Pod template.
+    let mut pb = PodBuilder::new();
+    // common pod settings
+    pb.metadata(ObjectMeta {
+        labels: Some(hdfs.rolegroup_selector_labels(rolegroup_ref)),
+        ..ObjectMeta::default()
+    })
+    .image_pull_secrets_from_product_image(resolved_product_image)
+    .add_volumes(vec![Volume {
+        name: "config".to_string(),
+        config_map: Some(ConfigMapVolumeSource {
+            name: Some(rolegroup_ref.object_name()),
+            ..ConfigMapVolumeSource::default()
+        }),
+        ..Volume::default()
+    }])
+    .service_account_name(rbac_sa)
+    .security_context(
+        PodSecurityContextBuilder::new()
+            .run_as_user(1000)
+            .run_as_group(1000)
+            .fs_group(1000) // Needed for secret-operator
+            .build(),
+    );
+
     let replicas;
-    let init_containers;
-    let containers;
 
     let (pvc, resources) = hdfs.resources(role, rolegroup_ref).unwrap_or_default();
 
+    // role specific pod settings configured here
     match role {
         HdfsRole::DataNode => {
-            replicas = hdfs.rolegroup_datanode_replicas(rolegroup_ref)?;
-            init_containers = datanode_init_containers(namenode_podrefs, hadoop_container);
-            containers = datanode_containers(rolegroup_ref, hadoop_container, &resources)?;
+            let rg = hdfs.datanode_rolegroup(rolegroup_ref);
+            replicas = rg.and_then(|rg| rg.replicas).unwrap_or_default();
+            pb.node_selector_opt(rg.and_then(|rg| rg.selector.clone()));
+            if let Some(init_containers) =
+                datanode_init_containers(namenode_podrefs, hadoop_container)
+            {
+                for c in init_containers {
+                    pb.add_init_container(c);
+                }
+            }
```

Comment:
It's a shame we don't have `PodBuilder::add_containers`
```suggestion
            for c in datanode_init_containers(namenode_podrefs, hadoop_container)
                .into_iter()
                .flatten()
            {
                pb.add_init_container(c);
            }
```

Reply:
agreed